### PR TITLE
IGNITE-16338 Remove usages of Cluster#isShutdown method

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-16338

### CODE REVIEW

File: modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java [hunk 1]
```diff
@@ -159,7 +168,12 @@ public void stop() {
                 }
 
                 cluster.shutdown();
-                cluster.onShutdown().block();
+
+                try {
+                    shutdownFuture.get(10, TimeUnit.SECONDS);
+                } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                    throw new IgniteInternalException("Unable to stop the ClusterService", e);
```
Comment (rpuch) [inline modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java:175]: Shouldn't we restore interruption status if we caught `InterruptedException` here using `Thread.interrupt()`?

File: modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java [hunk 2]
```diff
@@ -159,7 +168,12 @@ public void stop() {
                 }
 
                 cluster.shutdown();
-                cluster.onShutdown().block();
+
+                try {
+                    shutdownFuture.get(10, TimeUnit.SECONDS);
```
Comment (rpuch) [inline modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java:173]: Why is it 10 seconds? Is it some estimate, or just a value that seems sufficient?

Comment (sashapolo) [reply modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java:173]: Just a value that seems sufficient. Do you have any other ideas?

Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java:173]: The shutdown procedure leaves the cluster, shuts down membership, gossip, metadata store and does not seem to wait for any long-running tasks (or executors, if I didn't miss anything). So 10 seconds seems to be a good start.

**Review by rpuch — APPROVED** [review]

**Pull request merged**